app/src/main/java/org/android/go/sopt/presentation/sign/SignInActivity.kt

```diff
@@ -41,15 +45,18 @@ class SignInActivity : AppCompatActivity() {
         binding.btnSignUp.setOnClickListener {
             moveToSignUp()
         }
-        binding.btnSignIn.setOnClickListener {
-            viewModel.signIn()
-            if (viewModel.isCompleteSign.value) {
-                showToast(getString(R.string.sign_in_success_message))
-                GSDataStore(this).isLogin = true
-                moveToHome()
-            } else
-                showToast(getString(R.string.sign_in_fail_message))
-        }
+    }
+
+    private fun collectData() {
+        viewModel.isCompleteSign.flowWithLifecycle(lifecycle).onEach { isCompleteSign ->
+            isCompleteSign?.let {
+                if (it) {
+                    showToast(getString(R.string.sign_in_success_message))
+                    moveToHome()
+                } else
+                    showToast(getString(R.string.sign_in_fail_message))
+            }
```

Comment:
```suggestion
            if (isCompleteSign == null) return@onEach
            if (isCompleteSign) {
                showToast(getString(R.string.sign_in_success_message))
                moveToHome()
            } else {
                showToast(getString(R.string.sign_in_fail_message))
            }
        }
```

가독성을 위해 들여쓰기를 좀 더 줄여볼 수 있을 것 같아요~